testing/logging/test_fixture.py

```diff
@@ -137,3 +137,68 @@ def test_caplog_captures_for_all_stages(caplog, logging_during_setup_and_teardow
 
     # This reaches into private API, don't use this type of thing in real tests!
     assert set(caplog._item.catch_log_handlers.keys()) == {"setup", "call"}
+
+
+def test_ini_controls_global_log_level(testdir):
+    testdir.makepyfile(
+        """
+        import pytest
+        import logging
+        def test_log_level_override(request, caplog):
+            plugin = request.config.pluginmanager.getplugin('logging-plugin')
+            assert plugin.log_level == logging.ERROR
+            logging.getLogger('catchlog').warning("WARNING message won't be shown")
+            logging.getLogger('catchlog').error("ERROR message will be shown")
+            assert 'WARNING' not in caplog.text
+            assert 'ERROR' in caplog.text
+    """
+    )
+    testdir.makeini(
+        """
+        [pytest]
+        log_level=ERROR
+    """
+    )
+
+    result = testdir.runpytest()
+    # make sure that that we get a '0' exit code for the testsuite
+    assert result.ret == 0
+
+
+def test_caplog_can_override_global_log_level(testdir):
+    testdir.makepyfile(
+        """
+        import pytest
+        import logging
+        def test_log_level_override(request, caplog):
+            logger = 'catchlog'
+            plugin = request.config.pluginmanager.getplugin('logging-plugin')
+            assert plugin.log_level == logging.WARNING
+
+            logging.getLogger(logger).info("INFO message won't be shown")
+
+            caplog.set_level(logging.INFO, logger)
+
+            with caplog.at_level(logging.DEBUG, logger):
+                logging.getLogger(logger).debug("DEBUG message will be shown")
```

Comment:
What happens If you raise after this line? Is the DEBUG message shown in the captured log output?

Reply:
Yes it will be. I've added another test to this effect

Reply:
IMO this is bug, but it is negligible. I think it would be better, if a new logging handler were used for caplog, whose level is modified in set_level/at_level.

Reply:
> the levels passed to `caplog.set/at_level` may have an impact on the output log messages in the case of a failing test

@thisch, could you explain a little more how this is a bug? To me, this is what you would want. If you temporarily set the level to `DEBUG` to run a function and that function raises an exception, I would want to see the `DEBUG` messages output. Perhaps it could be a new feature to separate out the various handlers since I'm unsure this issue has been created by this PR.

Reply:
I have a different opinion about this. I think that a user who specifies `--log-level INFO`, would be surprised if the output (stdout) captured logs of a failing test, that uses `caplog.set_level(logging.DEBUG)`, contained `DEBUG` log messages.

As I said in https://github.com/pytest-dev/pytest/pull/7139#discussion_r419072623, I think that manually specified pytest logging settings on the cmd line or in `pytest.ini` must not lead to failing tests.

Reply:
Ok, if i understand correctly, this is about separating out the logs that are printed to the log report upon failure from the logs that `caplog` sees? This feels like a new piece of work so I would be happy to tackle this but in a separate PR as I don't think this was previously possible with `pytest` anyway?

Reply:
Let me know if this is what you had in mind and I can either include it here or raise a separate PR.

https://github.com/flexciton/pytest/commit/844259ac58817e740e0817ff4e7796d440b68e85

Reply:
> Ok, if i understand correctly, this is about separating out the logs that are printed to the log report upon failure from the logs that `caplog` sees?

Exactly!

> This feels like a new piece of work so I would be happy to tackle this but in a separate PR as I don't think this was previously possible with pytest anyway?

Tackling this in a separate PR is fine with me.

Reply:
> Let me know if this is what you had in mind and I can either include it here or raise a separate PR.
> 
> [flexciton@844259a](https://github.com/flexciton/pytest/commit/844259ac58817e740e0817ff4e7796d440b68e85)

Great. Thx for starting working on this! As I said before this should be moved to a separate PR.

Reply:
FYI raised #7159 to continue this work